test/unit/puppetlabs/services/jruby/jruby_testutils.clj

```diff
@@ -29,6 +28,26 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; JRubyPuppet Test util functions
 
+(schema/defn ^:always-validate
+create-mock-pool-instance :- JRubyInstance
```

Comment:
The part about it failing more often when run with other tests makes me wonder about whether some of the other tests aren't cleaning up fully before they exit in some cases.  I've definitely seen some weird things (that I can't really explain) with `with-redefs` which affect functions that run on different threads when the lifetime of code running on those threads spans thread boundaries.

I'm good with how this is implemented on this PR, though, so not suggesting we spend any more time diagnosing that at this point.